File: forms/agile-checklist/front-end-form-with-html/js/items.js
// Public exports attached to window.AgileChecklist.
(function () {
  'use strict';

  const TEAMS_ITEMS = [
    { id: 't01', ordinal: 1, section: 'teams', text: 'Teams have problems to solve rather than lists of tasks to perform.' },
    { id: 't02', ordinal: 2, section: 'teams', text: "Teams generally make decisions without a manager's approval." },
    { id: 't03', ordinal: 3, section: 'teams', text: 'Teams adopt appropriate practices and improve them over time.' },
    { id: 't04', ordinal: 4, section: 'teams', text: 'Teams working on a common product actively coordinate.' },
    { id: 't05', ordinal: 5, section: 'teams', text: 'Teams openly share new ideas and experiences with others.' },
    { id: 't06', ordinal: 6, section: 'teams', text: 'Teams decide how to execute work.' },
    { id: 't07', ordinal: 7, section: 'teams', text: 'Teams quickly act on stakeholder feedback to improve the product.' },
    { id: 't08', ordinal: 8, section: 'teams', text: 'Teams rarely wait for work to be completed by others.' },
    { id: 't09', ordinal: 9, section: 'teams', text: 'Teams make every effort to fully complete all planned work.' },
    { id: 't10', ordinal: 10, section: 'teams', text: 'Teams monitor and manage their own performance.' },
    { id: 't11', ordinal: 11, section: 'teams', text: 'Teams understand the benefits and challenges of agile.' },
    { id: 't12', ordinal: 12, section: 'teams', text: 'Teams responsibly deliver high quality work.' },
    { id: 't13', ordinal: 13, section: 'teams', text: 'Teams welcome rather than resist changing requirements.' },
    { id: 't14', ordinal: 14, section: 'teams', text: 'Teams collaborate to finish items.' },
    { id: 't15', ordinal: 15, section: 'teams', text: 'Teams admit challenges and mistakes.' },
    { id: 't16', ordinal: 16, section: 'teams', text: 'Teams can work outside their specialties to achieve goals.' },
    { id: 't17', ordinal: 17, section: 'teams', text: 'Teams seek to learn relevant, new skills.' },
    { id: 't18', ordinal: 18, section: 'teams', text: 'Teams continue to learn and improve skills.' },
    { id: 't19', ordinal: 19, section: 'teams', text: 'Teams actively seek ways to improve ways of working.' },
    { id: 't20', ordinal: 20, section: 'teams', text: 'Teams choose and use various ways of communicating.' },
    { id: 't21', ordinal: 21, section: 'teams', text: 'Teams have all received basic training on agile concepts.' },
    { id: 't22', ordinal: 22, section: 'teams', text: 'Teams feel safe to express dissenting views.' },
    { id: 't23', ordinal: 23, section: 'teams', text: 'Teams start work even when some open issues remain.' },
    { id: 't24', ordinal: 24, section: 'teams', text: 'Teams are motivated by performing their work.' },
    { id: 't25', ordinal: 25, section: 'teams', text: 'Teams take pride in their craft.' },
  ];

  const STAKEHOLDERS_ITEMS = [
    { id: 's01', ordinal: 1, section: 'stakeholders', text: 'Stakeholders know which factors determine priorities and deadlines.' },
    { id: 's02', ordinal: 2, section: 'stakeholders', text: 'Stakeholders accept plans that are expressed with ranges.' },
    { id: 's03', ordinal: 3, section: 'stakeholders', text: 'Stakeholders and the team accept that plans may need to change.' },
    { id: 's04', ordinal: 4, section: 'stakeholders', text: 'Stakeholders have frequent opportunities to evaluate the product.' },
    { id: 's05', ordinal: 5, section: 'stakeholders', text: 'Stakeholders champion agile values and agile principles.' },
    { id: 's06', ordinal: 6, section: 'stakeholders', text: 'Stakeholders see quality as a right that the team has and they respect it.' },
    { id: 's07', ordinal: 7, section: 'stakeholders', text: 'Stakeholders delegate authority to teams.' },
    { id: 's08', ordinal: 8, section: 'stakeholders', text: "Stakeholders don't take authority back at the first sign of trouble." },
    { id: 's09', ordinal: 9, section: 'stakeholders', text: 'Stakeholders support teams in experimenting.' },
    { id: 's10', ordinal: 10, section: 'stakeholders', text: "Stakeholders don't punish an unsuccessful experiment." },
    { id: 's11', ordinal: 11, section: 'stakeholders', text: 'Stakeholders communicate what they hope to achieve by becoming more agile.' },
    { id: 's12', ordinal: 12, section: 'stakeholders', text: 'Stakeholders actively encourage teams to learn new skills.' },
    { id: 's13', ordinal: 13, section: 'stakeholders', text: 'Stakeholders actively encourage teams to learn new ways of working.' },
    { id: 's14', ordinal: 14, section: 'stakeholders', text: 'Stakeholders develop people rather than manage their performance.' },
  ];

  const PRACTICES_ITEMS = [
    { id: 'p01', ordinal: 1, section: 'practices', text: 'The early release of a good product is generally favoured over the later release of a perfect product.' },
    { id: 'p02', ordinal: 2, section: 'practices', text: 'An educated executive sponsor ensures the change agent is empowered to make decisions.' },
    { id: 'p03', ordinal: 3, section: 'practices', text: 'Decisions are made quickly, sometimes with incomplete knowledge.' },
    { id: 'p04', ordinal: 4, section: 'practices', text: 'Plans are based on data and experience.' },
    { id: 'p05', ordinal: 5, section: 'practices', text: 'Dependencies between teams are proactively identified and addressed.' },
    { id: 'p06', ordinal: 6, section: 'practices', text: 'People are assumed to have acted with good intentions even when something goes wrong.' },
    { id: 'p07', ordinal: 7, section: 'practices', text: 'Trust is reciprocal within the team and with stakeholders.' },
    { id: 'p08', ordinal: 8, section: 'practices', text: 'Although formal documentation may exist, it is supplemented by conversations to the extent possible.' },
    { id: 'p09', ordinal: 9, section: 'practices', text: 'As more is learned, plans are updated rather than remaining based on outdated information.' },
    { id: 'p10', ordinal: 10, section: 'practices', text: "A non-punitive attitude leads to the team's willingness to experiment with new tools, technologies, and ways of working." },
    { id: 'p11', ordinal: 11, section: 'practices', text: 'Outside groups that partner with the team are aware of how this new way of working may affect their interactions.' },
    { id: 'p12', ordinal: 12, section: 'practices', text: 'The organisation places a higher value on finished work than it does on the number of work items in process.' },
    { id: 'p13', ordinal: 13, section: 'practices', text: 'Quality is rarely, if ever, sacrificed to meet a deadline.' },
    { id: 'p14', ordinal: 14, section: 'practices', text: 'When problems arise between teams, everyone focuses on a solution rather than blame.' },
    { id: 'p15', ordinal: 15, section: 'practices', text: 'Change agents are in place to help support and guide the transition.' },
    { id: 'p16', ordinal: 16, section: 'practices', text: 'Agile practices are being implemented beyond the group in which agile began.' },
    { id: 'p17', ordinal: 17, section: 'practices', text: 'Most people are assigned to only one team.' },
    { id: 'p18', ordinal: 18, section: 'practices', text: 'People honour commitments and keep promises.' },
  ];

  const ALL_ITEMS = TEAMS_ITEMS.concat(STAKEHOLDERS_ITEMS).concat(PRACTICES_ITEMS);

  const SECTION_LABEL = {
    teams: 'Teams',
    stakeholders: 'Stakeholders',
    practices: 'Practices',
  };

  window.AgileChecklist = window.AgileChecklist || {};
  window.AgileChecklist.TEAMS_ITEMS = TEAMS_ITEMS;
  window.AgileChecklist.STAKEHOLDERS_ITEMS = STAKEHOLDERS_ITEMS;
  window.AgileChecklist.PRACTICES_ITEMS = PRACTICES_ITEMS;
  window.AgileChecklist.ALL_ITEMS = ALL_ITEMS;
  window.AgileChecklist.SECTION_LABEL = SECTION_LABEL;
})();

File: forms/agile-checklist/back-end-with-loco/src/engine/items.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub enum Section {
    Teams,
    Stakeholders,
    Practices,
}

impl Section {
    pub fn slug(self) -> &'static str {
        match self {
            Section::Teams => "teams",
            Section::Stakeholders => "stakeholders",
            Section::Practices => "practices",
        }
    }
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ItemDef {
    pub id: &'static str,
    pub ordinal: u8,
    pub section: Section,
    pub text: &'static str,
}

macro_rules! item {
    ($id:expr, $ord:expr, $sec:expr, $text:expr) => {
        ItemDef { id: $id, ordinal: $ord, section: $sec, text: $text }
    };
}

pub const TEAMS_ITEMS: &[ItemDef] = &[
    item!("t01", 1, Section::Teams, "Teams have problems to solve rather than lists of tasks to perform."),
    item!("t02", 2, Section::Teams, "Teams generally make decisions without a manager's approval."),
    item!("t03", 3, Section::Teams, "Teams adopt appropriate practices and improve them over time."),
    item!("t04", 4, Section::Teams, "Teams working on a common product actively coordinate."),
    item!("t05", 5, Section::Teams, "Teams openly share new ideas and experiences with others."),
    item!("t06", 6, Section::Teams, "Teams decide how to execute work."),
    item!("t07", 7, Section::Teams, "Teams quickly act on stakeholder feedback to improve the product."),
    item!("t08", 8, Section::Teams, "Teams rarely wait for work to be completed by others."),
    item!("t09", 9, Section::Teams, "Teams make every effort to fully complete all planned work."),
    item!("t10", 10, Section::Teams, "Teams monitor and manage their own performance."),
    item!("t11", 11, Section::Teams, "Teams understand the benefits and challenges of agile."),
    item!("t12", 12, Section::Teams, "Teams responsibly deliver high quality work."),
    item!("t13", 13, Section::Teams, "Teams welcome rather than resist changing requirements."),
    item!("t14", 14, Section::Teams, "Teams collaborate to finish items."),
    item!("t15", 15, Section::Teams, "Teams admit challenges and mistakes."),
    item!("t16", 16, Section::Teams, "Teams can work outside their specialties to achieve goals."),
    item!("t17", 17, Section::Teams, "Teams seek to learn relevant, new skills."),
    item!("t18", 18, Section::Teams, "Teams continue to learn and improve skills."),
    item!("t19", 19, Section::Teams, "Teams actively seek ways to improve ways of working."),
    item!("t20", 20, Section::Teams, "Teams choose and use various ways of communicating."),
    item!("t21", 21, Section::Teams, "Teams have all received basic training on agile concepts."),
    item!("t22", 22, Section::Teams, "Teams feel safe to express dissenting views."),
    item!("t23", 23, Section::Teams, "Teams start work even when some open issues remain."),
    item!("t24", 24, Section::Teams, "Teams are motivated by performing their work."),
    item!("t25", 25, Section::Teams, "Teams take pride in their craft."),
];

pub const STAKEHOLDERS_ITEMS: &[ItemDef] = &[
    item!("s01", 1, Section::Stakeholders, "Stakeholders know which factors determine priorities and deadlines."),
    item!("s02", 2, Section::Stakeholders, "Stakeholders accept plans that are expressed with ranges."),
    item!("s03", 3, Section::Stakeholders, "Stakeholders and the team accept that plans may need to change."),
    item!("s04", 4, Section::Stakeholders, "Stakeholders have frequent opportunities to evaluate the product."),
    item!("s05", 5, Section::Stakeholders, "Stakeholders champion agile values and agile principles."),
    item!("s06", 6, Section::Stakeholders, "Stakeholders see quality as a right that the team has and they respect it."),
    item!("s07", 7, Section::Stakeholders, "Stakeholders delegate authority to teams."),
    item!("s08", 8, Section::Stakeholders, "Stakeholders don't take authority back at the first sign of trouble."),
    item!("s09", 9, Section::Stakeholders, "Stakeholders support teams in experimenting."),
    item!("s10", 10, Section::Stakeholders, "Stakeholders don't punish an unsuccessful experiment."),
    item!("s11", 11, Section::Stakeholders, "Stakeholders communicate what they hope to achieve by becoming more agile."),
    item!("s12", 12, Section::Stakeholders, "Stakeholders actively encourage teams to learn new skills."),
    item!("s13", 13, Section::Stakeholders, "Stakeholders actively encourage teams to learn new ways of working."),
    item!("s14", 14, Section::Stakeholders, "Stakeholders develop people rather than manage their performance."),
];

pub const PRACTICES_ITEMS: &[ItemDef] = &[
    item!("p01", 1, Section::Practices, "The early release of a good product is generally favoured over the later release of a perfect product."),
    item!("p02", 2, Section::Practices, "An educated executive sponsor ensures the change agent is empowered to make decisions."),
    item!("p03", 3, Section::Practices, "Decisions are made quickly, sometimes with incomplete knowledge."),
    item!("p04", 4, Section::Practices, "Plans are based on data and experience."),
    item!("p05", 5, Section::Practices, "Dependencies between teams are proactively identified and addressed."),
    item!("p06", 6, Section::Practices, "People are assumed to have acted with good intentions even when something goes wrong."),
    item!("p07", 7, Section::Practices, "Trust is reciprocal within the team and with stakeholders."),
    item!("p08", 8, Section::Practices, "Although formal documentation may exist, it is supplemented by conversations to the extent possible."),
    item!("p09", 9, Section::Practices, "As more is learned, plans are updated rather than remaining based on outdated information."),
    item!("p10", 10, Section::Practices, "A non-punitive attitude leads to the team's willingness to experiment with new tools, technologies, and ways of working."),
    item!("p11", 11, Section::Practices, "Outside groups that partner with the team are aware of how this new way of working may affect their interactions."),
    item!("p12", 12, Section::Practices, "The organisation places a higher value on finished work than it does on the number of work items in process."),
    item!("p13", 13, Section::Practices, "Quality is rarely, if ever, sacrificed to meet a deadline."),
    item!("p14", 14, Section::Practices, "When problems arise between teams, everyone focuses on a solution rather than blame."),
    item!("p15", 15, Section::Practices, "Change agents are in place to help support and guide the transition."),
    item!("p16", 16, Section::Practices, "Agile practices are being implemented beyond the group in which agile began."),
    item!("p17", 17, Section::Practices, "Most people are assigned to only one team."),
    item!("p18", 18, Section::Practices, "People honour commitments and keep promises."),
];

pub fn all_items() -> Vec<&'static ItemDef> {
    TEAMS_ITEMS
        .iter()
        .chain(STAKEHOLDERS_ITEMS.iter())
        .chain(PRACTICES_ITEMS.iter())
        .collect()
}
